### PR TITLE
Add a gcloud-terraform image and corresponding publishing job.

### PR DESCRIPTION
**Pull request by cjwagner**
/assign @chaodaiG 

I think this has all the utilities we'll need for validating terraform config and applying. I've confirmed that the Dockerfile builds locally.

### CODE REVIEW

File: images/gcloud-terraform/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 The Kubernetes Authors.
```
Comment (chaodaiG) [inline images/gcloud-terraform/Dockerfile:1]: 2021

File: images/gcloud-terraform/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    /google-cloud-sdk/install.sh \
+        --disable-installation-options \
+        --bash-completion=false \
+        --path-update=false \
+        --usage-reporting=false && \
+    gcloud components install alpha beta kubectl && \
+    gcloud info | tee /workspace/gcloud-info.txt
+
+# Install terraform
+RUN apk add --no-cache terraform
```
Comment (chaodaiG) [inline images/gcloud-terraform/Dockerfile:49]: can we add a version here?

Comment (cjwagner) [reply images/gcloud-terraform/Dockerfile]: Done (although I had to pick a newer version than the one specified in the README due to availability).

File: images/gcloud-terraform/Dockerfile (hunk 1 above)
Comment (cjwagner) [reply images/gcloud-terraform/Dockerfile]: Fixed.

**Review by chaodaiG — APPROVED** [review]

**Pull request merged**